Source/WebCore/testing/Internals.cpp

```diff
@@ -3951,12 +3963,13 @@ void Internals::setFullscreenInsets(FullscreenInsets insets)
     page->setFullscreenInsets(FloatBoxExtent(insets.top, insets.right, insets.bottom, insets.left));
 }
 
-void Internals::setFullscreenAutoHideDuration(double duration)
+ExceptionOr<void> Internals::setFullscreenAutoHideDuration(double duration)
 {
-    Page* page = contextDocument()->frame()->page();
-    ASSERT(page);
-
-    page->setFullscreenAutoHideDuration(Seconds(duration));
+    RefPtr document = contextDocument();
+    if (!document || !document->page())
+        return Exception { ExceptionCode::InvalidStateError };
+    document->page()->setFullscreenAutoHideDuration(Seconds(duration));
```

Comment:
I’d think this would need to be in a `RefPtr` or call `protectedFrame()` in our new safer C++ model.

Reply:
Done